src/_pytest/logging.py

```diff
@@ -468,6 +469,16 @@ def _setup_cli_logging(self):
             log_cli_handler, formatter=log_cli_formatter, level=log_cli_level
         )
 
+    def set_log_path(self, fname):
+        if isinstance(fname, pathlib.PurePath):
```

Comment:
I think just `fname = Path(fname)` is enough instead of this check. 👍

Reply:
Sorry, did not understand why to cast vica versa? 
At the end we will need `str` type for `fname` to work correctly within `logging.FileHandler()`.
`fname` parameter can be `str`, or `pathlib` type. If it is `str` type everything is ok, `logging.FileHandler()` will work. But if it is `pathlib` type I would cast it to `str` at beginning of this function.
Please correct me if I am wrong. Thanks

Reply:
Sorry, Just read: https://docs.python.org/3/library/logging.handlers.html#logging.FileHandler, 
From 3.6 it supports Path() also. But I have an exception if filename in `pathlib` type

Reply:
Maybe for compatibility reasons it is ok if the type is str

Reply:
we should use pathlib and cast to str until we drop suport for python < 3.6

Reply:
Thanks, fixed